swo_aws_extension/flows/jobs/billing_journal/billing_journal_generator.py

```diff
@@ -710,3 +707,117 @@ def _get_services_amounts_by_account(account_reports):
                     services_by_account[service_name] = amount
 
         return services_by_account
+
+    def _get_exchange_rate_by_invoice_entity_and_currency(
+        self, organization_invoices, invoice_entity
+    ):
+        """Gets the maximum exchange rate for a specific invoicing entity and currency
+        from a list of organization invoice summaries. If no invoices are found for the
+        specified invoicing entity, it returns the maximum exchange rate across all invoices
+        for the given currency.
+
+        Args:
+            organization_invoices (list): List of organization invoice summaries.
+            invoice_entity (str): The invoicing entity to filter by.
+
+        Returns:
+            float: The maximum exchange rate for the specified invoicing entity,
+            or the maximum exchange rate across all invoices if none found.
+        """
+        exchange_rates = [
+            float(
+                inv.get("PaymentCurrencyAmount", {})
+                .get("CurrencyExchangeDetails", {})
+                .get("Rate", 0)
+            )
+            for inv in organization_invoices
+            if inv.get("Entity", {}).get("InvoicingEntity") == invoice_entity
+            and inv.get("PaymentCurrencyAmount", {}).get("CurrencyCode", "")
+            == self.authorization_currency
+        ]
+        invoice_entity_exchange_rate = max(exchange_rates) if exchange_rates else 0
+
+        if invoice_entity_exchange_rate == 0:
+            exchange_rates = [
+                float(
+                    inv.get("PaymentCurrencyAmount", {})
+                    .get("CurrencyExchangeDetails", {})
+                    .get("Rate", 0)
+                )
+                for inv in organization_invoices
+                if inv.get("PaymentCurrencyAmount", {}).get("CurrencyCode", "")
+                == self.authorization_currency
+            ]
+            return max(exchange_rates) if exchange_rates else 0
```

Comment:
https://docs.python.org/3/library/functions.html#max

```suggestion
            return max(exchange_rates, default=0)
```